roachtest: acceptance/version-upgrade fails due to panic in schema changer

Found in CI for an unrelated PR: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_Roachtest/2529759?buildTab=artifacts

```
run_174620.242_n1_workload_run_schemachange: 17:46:20 cluster.go:386: > /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:1 -- ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 10 --concurrency 2 {pgurl:1-4}
I201218 17:46:20.327435 1 workload/cli/run.go:359  creating load generator...
I201218 17:46:20.730447 1 workload/cli/run.go:390  creating load generator... done (took 403.005703ms)
BEGIN
  CREATE TYPE public.enum19 AS ENUM ('afsaas');
ROLLBACK; expected SQLSTATE(S) 42710, and got ERROR: type "public.enum19" already exists (SQLSTATE 42710)
BEGIN
  ALTER TABLE public.table7 RENAME TO public.table7;
  CREATE TABLE public.table10 AS SELECT public.table7.col7_12, public.table7.col7_10, public.table7.col7_14, public.table7.col7_16, public.table7.col7_13, public.table7.col7_9, public.table7.col7_15, public.table7.col7_18, public.table7.col7_8, public.table7.col7_11, public.table7.col7_17, public.table7.col7_12 FROM public.table7, public.table7;
ROLLBACK; expected SQLSTATE(S) 42701,42712, and got ERROR: source name "table7" specified more than once (missing AS clause) (SQLSTATE 42712)
BEGIN
  ALTER TABLE public.table7 ALTER COLUMN "col7_9" SET NOT NULL;
COMMIT; 
BEGIN
  CREATE SEQUENCE IF NOT EXISTS public.seq11;
  ALTER TABLE public.table7 ALTER COLUMN "col7_8" DROP STORED;
ROLLBACK; expected SQLSTATE(S) 42611, and got ERROR: column "col7_8" is not a computed column (SQLSTATE 42611)
BEGIN
  INSERT INTO public.table7 (col7_8,col7_9,col7_10,col7_11,col7_12,col7_13,col7_14,col7_15,col7_16,col7_17,col7_18) VALUES (1446961652:::OID,e'FK(\x04EX;?':::STRING:::NAME,'1989-11-05 21:03:05.000583':::TIMESTAMP,19109:::INT8,'=z':::STRING:::NAME,B'01100001101010100010100001000001100','1988-09-18':::DATE,(-1688783588827393420):::INT8,'1983-03-08 23:20:36.000754+00:00':::TIMESTAMPTZ,871774598:::OID,'\x58':::BYTES);
  ALTER TABLE public.table7 DROP COLUMN "col7_9";
ROLLBACK; expected SQLSTATE(S) 0A000,42P10, and got ERROR: unimplemented: schema change statement cannot follow a statement that has written in the same transaction (SQLSTATE 0A000)
BEGIN
  CREATE VIEW public.view12 AS SELECT public.table7.col7_14, public.table7.col7_12 FROM public.table7;
  DROP TABLE public.table7 RESTRICT;
ROLLBACK; expected SQLSTATE(S) 2BP01, and got ERROR: cannot drop relation "table7" because view "view12" depends on it (SQLSTATE 2BP01)



panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x184e299]

goroutine 36 [running]:
panic(0x1e39e40, 0x51aa690)
	/usr/local/go/src/runtime/panic.go:1064 +0x545 fp=0xc000fc7460 sp=0xc000fc7398 pc=0x4504a5
runtime.panicmem(...)
	/usr/local/go/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:742 +0x413 fp=0xc000fc7490 sp=0xc000fc7460 pc=0x467073
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).Family(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1124
github.com/cockroachdb/cockroach/pkg/sql/rowenc.RandDatumWithNullChance(0xc00103d0b0, 0x0, 0x0, 0x7, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowenc/testutils.go:83 +0x79 fp=0xc000fc77d0 sp=0xc000fc7490 pc=0x184e299
github.com/cockroachdb/cockroach/pkg/sql/rowenc.RandDatum(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowenc/testutils.go:63
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).setColumnDefault(0xc00103d1d0, 0xc00039eb70, 0x18, 0xc0001691d0, 0xc, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:1393 +0x273 fp=0xc000fc7960 sp=0xc000fc77d0 pc=0x1879053
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp(0xc00103d1d0, 0xc00039eb70, 0x2, 0x5, 0x0, 0x7f065245d4f8, 0x20, 0x28)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:230 +0x20b fp=0xc000fc7a78 sp=0xc000fc7960 pc=0x186edeb
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn(0xc00103d230, 0xc00039eb70, 0x0, 0x0, 0xc00039eb70, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:239 +0xff fp=0xc000fc7c38 sp=0xc000fc7a78 pc=0x1882fff
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run(0xc00103d230, 0x3fcd640, 0xc00037f680, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:322 +0x125 fp=0xc000fc7ea8 sp=0xc000fc7c38 pc=0x1884605
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run-fm(0x3fcd640, 0xc00037f680, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:313 +0x3e fp=0xc000fc7ee0 sp=0xc000fc7ea8 pc=0x1887b7e
github.com/cockroachdb/cockroach/pkg/workload/cli.workerRun(0x3fcd640, 0xc00037f680, 0xc000366360, 0xc000d86cb0, 0x0, 0xc00021ce30)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:237 +0xaf fp=0xc000fc7f58 sp=0xc000fc7ee0 pc=0x13f46ef
github.com/cockroachdb/cockroach/pkg/workload/cli.runRun.func2.1(0xc00098e000, 0xc00099e080, 0xc000366360, 0x0, 0x3fcd640, 0xc00037f680, 0xc000d86cb0, 0x1, 0xc00021ce30)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:426 +0xec fp=0xc000fc7f98 sp=0xc000fc7f58 pc=0x13f868c
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc000fc7fa0 sp=0xc000fc7f98 pc=0x4890e1
created by github.com/cockroachdb/cockroach/pkg/workload/cli.runRun.func2
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:416 +0x111

goroutine 1 [select]:
```

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -1629,6 +1629,9 @@ func (og *operationGenerator) getTableColumns(
 	if err := rows.Err(); err != nil {
 		return nil, err
 	}
+	if len(ret) == 0 {
+		return nil, pgx.ErrNoRows
+	}
 	for i := range ret {
 		c := &ret[i]
 		c.typ, err = og.typeFromTypeName(tx, typNames[i])

--- a/pkg/workload/schemachange/schemachange.go
+++ b/pkg/workload/schemachange/schemachange.go
@@ -173,9 +173,20 @@ func (s *schemaChange) initSeqNum(pool *workload.MultiConnPool) (*int64, error) 
 	seqNum := new(int64)
 
 	const q = `
-SELECT max(regexp_extract(name, '[0-9]+$')::int)
-  FROM ((SELECT table_name FROM [SHOW TABLES]) UNION (SELECT sequence_name FROM [SHOW SEQUENCES])) AS obj(name)
- WHERE name ~ '^(table|view|seq)[0-9]+$';
+SELECT max(regexp_extract(name, '[0-9]+$')::INT8)
+  FROM (
+    SELECT name
+      FROM (
+	           (SELECT table_name FROM [SHOW TABLES]) UNION
+						 (SELECT sequence_name FROM [SHOW SEQUENCES]) UNION
+						 (SELECT name FROM [SHOW ENUMS]) UNION
+	           (SELECT schema_name FROM [SHOW SCHEMAS]) UNION
+						 (SELECT column_name FROM information_schema.columns) UNION
+						 (SELECT index_name FROM information_schema.statistics)
+           ) AS obj (name)
+       )
+ WHERE name ~ '^(table|view|seq|enum|schema)[0-9]+$'
+    OR name ~ '^(col|index)[0-9]+_[0-9]+$';
 `
 	var max gosql.NullInt64
 	if err := pool.Get().QueryRow(q).Scan(&max); err != nil {

